kpi/serializers/v2/asset.py

```diff
@@ -343,6 +343,41 @@ class Meta(AssetSerializer.Meta):
                   'downloads',
                   )
 
+    def get_permissions(self, obj):
+        try:
+            # TODO refactored this code if possible.
+            # Quite redundant with `ObjectPermissionHelper.get_assignments_queryset`
+            assignments = self.context[
+                'object_permissions_per_object'].get(obj.pk)
+            context = self.context
+            request = self.context.get('request')
+            user_assignments = []
+            filtered_user_ids = None
+            user = request.user
+
+            if not user or user.is_anonymous():
+                filtered_user_ids = [obj.owner_id]
+            elif not ObjectPermissionHelper.user_can_share(obj, user):
+                # Display only users' permissions if they are not allowed to modify
+                # others' permissions
+                filtered_user_ids = [obj.owner_id, user.pk]
+
+            for assignment in assignments:
+                if (filtered_user_ids is None or
+                        assignment.user_id in filtered_user_ids):
+                    user_assignments.append(assignment)
+
+            context.update({'asset': obj})
+            context.update({'asset_uid': obj.uid})
+
+            return AssetPermissionAssignmentSerializer(user_assignments,
+                                                       many=True, read_only=True,
+                                                       context=context).data
+        except KeyError:
```

Comment:
It looks like `self.context['object_permissions_per_object']` is the only lookup by key, so can we `except` right after that? If the two possible execution paths are

1. finding `'object_permissions_per_object'` in `self.context` and eventually returning an instance of `AssetPermissionAssignmentSerializer`, and
2. _not_ finding `'object_permissions_per_object'` and punting to the superclass' `get_permissions()`,

then we should be able to `return super(AssetListSerializer, self).get_permissions(obj)` in the `except` right after `try`ing `self.context['object_permissions_per_object']`.

It's not purely a stylistic thing: what if a `KeyError` was raised in `user_can_share()` or the constructor for `AssetPermissionAssignmentSerializer` due to a programming error? It would be discarded silently.